src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -406,21 +406,21 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
   // Execution table.
   override def updateExecutionBackendInfo(workflowId: WorkflowId,
                                           call: Call,
+                                          index: Option[Int],
                                           backendInfo: CallBackendInfo): Future[Unit] = {
     require(backendInfo != null, "backend info is null")
     val callStatus = backendInfo.status
 
     val action = for {
-      executionResult <- dataAccess.executionsByWorkflowExecutionUuidAndCallFqn(
-        workflowId.toString, call.fullyQualifiedName).result.head
+      executionResult <- if (index.isDefined){
```

Comment:
You should be able to get both branches with some `ExecutionIndex` implicit magic:

```scala
import ExecutionIndex._

executionResult <- dataAccess.executionsByWorkflowExecutionUuidAndCallFqnAndShardIndex(workflowId.toString, call.fullyQualifiedName, index.fromIndex).result.head
```